content/recipes/troubleshooting-your-solution.md

```diff
@@ -0,0 +1,364 @@
+
+
+
+<a name="my-machine"> <a name="works-but-no">
+### My solution works on my machine, but not on Codewars, ***OR*** My solution seems correct, but some tests fail when I attempt it.
+</a></a>
+
+There are a number of things it could be. Try following (in given order):
+* Make sure you didn't miss anything in the the kata description. It's true that some descriptions are far from perfect, but it's also very likely you missed some requirement, did not account for some edge case, or that there's some helpful hint or advice there.
+
+* Make sure your solution is efficient enough. If you're setting a timeout error, you may need to rethink your approach (or check for infinite loops on edge cases!).
+
+* Some tests are susceptible to mutation of input parameters and fail if solution changes input objects passed to it. If kata does not ask you explicitly to change input arguments, make sure you do not do it. If you find a kata which fails when input is mutated by solution, raise it as an issue ([see here](#post-discourse)).
+
+* **Debug your solution.** Check what input causes your solution to fail ([see here](#print-input)), recreate it locally, and debug it in your IDE.
+
+* Check if your solution passes two successive calls with different inputs. It's sometimes the case that solution contains some global variables or static members and stale state is carried over from one invocation to another.
+
+* Check version of the language you have installed locally and what language version (or runtime) is used in Codewars runner. Maybe version installed on Codewars does not support some functions?
+
+* Read through this FAQ once again and search for a point which describes your problem better than “it does not work”.
+
+* Ask a question ([see here](#post-discourse)).
+
+
+
+
+
+<a name="test-attempt">
+
+### When I run "Tests", everything passes, but when I hit “Attempt” tests fail
+
+</a>
+
+It’s because your solution does not pass the tests. Remember that when you press **“TEST”**, only sample tests are run - the ones you see under the kata code editor. There’s usually only a couple of sample tests, they are easy, execute fast, and they do not cover too many edge cases. Only when you press **“ATTEMPT”**, the full test suite is run, potentially with tests for edge cases, large inputs, performance, random tests, etc. You cannot see full test suite (and its test cases) if you haven’t solved the kata. Only when full test suite is run and passes, your solution can be accepted and kata can be considered solved. See [this point](#works-but-no) for possible hints how to proceed.
+
+
+
+
+
+<a name="last-fail">
+
+### All tests except the last one pass
+
+</a>
+
+For some kata (but not for all!), test suite is built in such a way that tests are stopped on first failure. When you see only one failed test and it’s the last one, it might mean that it’s the first one which your solution failed, and when you fix it, there are more to come.
```

Comment:
```suggestion
<!--- For false positive "built in" -->
<!--- textlint-disable terminology -->

For some kata (but not for all!), test suite is built in such a way that tests are stopped on first failure. When you see only one failed test and it’s the last one, it might mean that it’s the first one which your solution failed, and when you fix it, there are more to come.

<!-- textlint-enable terminology -->
```